notify/webhook/webhook.go

```diff
@@ -100,8 +100,30 @@ func (n *Notifier) Notify(ctx context.Context, alerts ...*types.Alert) (bool, er
 	}
 
 	var buf bytes.Buffer
-	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
-		return false, err
+	if n.conf.Json != "" {
+		body, err := n.tmpl.ExecuteTextString(n.conf.Json, data)
+		if err != nil {
+			return false, err
+		}
+		if !json.Valid([]byte(body)) {
+			return false, fmt.Errorf("Json格式异常: %s ", body)
+		}
+		if err := json.Indent(&buf, []byte(body), "", "  "); err != nil {
+			return false, fmt.Errorf("Json格式异常: %s ", body)
+		}
+	} else if n.conf.Text != "" {
+		body, err := n.tmpl.ExecuteTextString(n.conf.Json, data)
+		if err != nil {
+			return false, err
+		}
+		err = json.NewEncoder(&buf).Encode(body)
+		if err != nil {
+			return false, err
+		}
```

Comment:
No need to json encode if this is Text template imho, "buf.WriteString(body)" should be sufficient

Reply:
> No need to json encode if this is Text template imho, "buf.WriteString(body)" should be sufficient

If neither the JSON template nor the text template is configured, then the message should be converted into JSON according to the original logic.